pkgs/development/python-modules/meep/default.nix

```diff
@@ -24,10 +24,9 @@
   numpy,
   scipy,
   matplotlib,
-  h5py-mpi,
+  h5py,
   cython,
   autograd,
-  mpi4py,
```

Comment:
meep works perfectly fine and we are actively using it with MPI. The calculations are heavy and you really want to have MPI support there. Please don't remove and don't break it.

Reply:
> meep works perfectly fine and we are actively using it with MPI. The calculations are heavy and you really want to have MPI support there. Please don't remove and don't break it.

Have you tested whether this is really breaking this? Could it be that in your current setup `meep` is using mpi directly and not via mpi4py?

Reply:
The `checkPhase` of it performs such a check that runs `mpiexec` directly.

Reply:
Yes, doesn't work without MPI. Using mpiexec is necessary to start the multiple processes and make them find each other in an MPI communicator. mpi4py may try to do it itself but usually this is a quite insane approach. So the presence of mpiexec doesn't have to say too much there. Still, mpi4py is required for the MPI API in Python.

Reply:
> Yes, doesn't work without MPI. Using mpiexec is necessary to start the multiple processes and make them find each other in an MPI communicator. mpi4py may try to do it itself but usually this is a quite insane approach. So the presence of mpiexec doesn't have to say too much there. Still, mpi4py is required for the MPI API in Python.

OK it came out good you insisted. Inspecting further the state of things upstream led me to realize mpi4py is not exactly broken, the issue was that I didn't add `mpi` to the `nativeBuildInputs`, and using the latest revision also helped.

I also managed to successfully use it to package a new `pytest-mpi`, and add tests to `h5py` (the later was the original motivation for this PR).